### PR TITLE
Add save_on_rank argument to DiskSaver and Checkpoint

### LINKED ISSUE #2635: Upgrading to 0.4.9 stuck multi-gpu training
## 🐛 Bug description

Hi @vfdev-5 ,

We upgraded ignite from 0.4.8 to 0.4.9 in MONAI 0.9.1 recently: https://github.com/Project-MONAI/MONAI/pull/4605.
Then got the issue report from user:
something changed related to multi-gpu training between 0.9.1 and 0.9.0... monailabel multi-training is not working.. SupervisedTrainer is getting stuck to run inference step to compute the loss.. after debugging a bit.. i see this is the problem... pytorch-ignite==0.4.8 vs pytorch-ignite==0.4.9 when I downgrade it, all is ok..

CC @wyli @SachidanandAlle
## Environment

- PyTorch Version (e.g., 1.4): 1.12.0
- Ignite Version (e.g., 0.3.0): 0.4.9
- OS (e.g., Linux): ubuntu
- How you installed Ignite (`conda`, `pip`, source): pip
- Python version: 3.8
- Any other relevant information: downgrade to 0.4.8 then everything goes fine

### PR DESCRIPTION
**Pull request by sadra-barikbin**
Related to #2635 

Description: Added `save_on_rank` to `DiskSaver`, `Checkpoint` and `ModelCheckpoint` classes.

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @sadra-barikbin 

However, I do not think this resolves https://github.com/pytorch/ignite/issues/2635 as it would also stuck on DiskSaver._check_and_setup in ctor.
